MSAL/src/instance/oauth2/b2c/MSALB2COauth2Provider.m

```diff
@@ -62,19 +63,25 @@ - (MSALResult *)resultWithTokenResult:(MSIDTokenResult *)tokenResult
     return [MSALResult resultWithMSIDTokenResult:tokenResult authority:b2cAuthority error:error];
 }
 
-- (MSIDAuthority *)issuerAuthorityWithAccount:(__unused MSALAccount *)account
+- (MSIDAuthority *)issuerAuthorityWithAccount:(MSALAccount *)account
                              requestAuthority:(MSIDAuthority *)requestAuthority
-                                        error:(__unused NSError **)error
+                                        error:(NSError **)error
 {
-    /*
-     In the acquire token silent call we assume developer wants to get access token for account's home tenant,
-     if authority is a common, organizations or consumers authority.
-     */
-    return [[MSIDB2CAuthority alloc] initWithURL:requestAuthority.url
-                                  validateFormat:NO
-                                       rawTenant:account.lookupAccountIdentifier.utid
-                                         context:nil
-                                           error:error];
+    if (self.accountMetadataCache)
+    {
+        NSURL *cachedURL = [self.accountMetadataCache getAuthorityURL:requestAuthority.url
+                                                        homeAccountId:account.homeAccountId.identifier
+                                                             clientId:self.clientId
+                                                              context:nil
+                                                                error:error];
+        if (!cachedURL) return requestAuthority;
+        return [[MSIDB2CAuthority alloc] initWithURL:cachedURL?:requestAuthority.url
```

Comment:
nit: cachedURL?:requestAuthority.url this will never be executed

Reply:
good catch :p